parquet/src/main/java/org/apache/iceberg/parquet/ParquetBloomRowGroupFilter.java

```diff
@@ -334,5 +335,10 @@ private <T> boolean shouldRead(
           return ROWS_MIGHT_MATCH;
       }
     }
+
+    @Override
+    public <T> Boolean handleNonReference(Bound<T> term) {
+      return ROWS_MIGHT_MATCH;
```

Comment:
Throw exceptions by default.

Reply:
Why does this need to change? Changing the Filter implementations just to implement constant folding in Spark in a big red flag.

Reply:
@rdblue this is not for constant folding. The transform function such as `years(ts) = 20` could be pushed to fileformat such as parquet/orc. The default behavior for those non-reference terms is throwing an exception.

Reply:
It sounds like this is a consequence of the last PR, #7886, but I don't see how it is related to the rewrites that are the purpose of this one. Can you move these changes to a separate PR to be reviewed?

Reply:
Submit #8243 for parquet and #8244 for orc. Will rebase after they are merged. Keep here for CI.

Reply:
Both of these were merged.

Reply:
Thanks, rebased